crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net_task.rs

```diff
@@ -0,0 +1,1157 @@
+    async fn make_peer_query_request() {
+        let (mut ep_evt_send, _, _, _, _, _) = setup().await;
+
+        let (send_res, read_res) = futures::channel::oneshot::channel();
+
+        ep_evt_send
+            .send(MetaNetEvt::Request {
+                remote_url: "".to_string(),
+                con: mk_test_con_with_id(1),
+                data: wire::Wire::PeerQuery(wire::PeerQuery {
+                    space: test_space(1),
+                    basis_loc: DhtLocation::new(1),
+                }),
+                respond: Box::new(|r| {
+                    async move {
+                        send_res.send(r).unwrap();
+                        ()
+                    }
+                    .boxed()
+                    .into()
+                }),
+            })
+            .await
+            .unwrap();
+
+        let response = tokio::time::timeout(Duration::from_secs(1), read_res)
+            .await
+            .expect("Timed out while waiting for a response")
+            .unwrap();
+
+        let peer_list = match response {
+            Wire::PeerQueryResp(r) => r.peer_list,
+            r => panic!("Unexpected response - {:?}", r),
+        };
+
+        assert_eq!(8, peer_list.len());
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn handle_peer_query_request_error() {
+        let (mut ep_evt_send, _, _, host_receiver_stub, _, _) = setup().await;
+
+        // Set up the error response so that when we make a request we get an error
+        host_receiver_stub
+            .respond_with_error
+            .store(true, Ordering::SeqCst);
+
+        let (send_res, read_res) = futures::channel::oneshot::channel();
+
+        ep_evt_send
+            .send(MetaNetEvt::Request {
+                remote_url: "".to_string(),
+                con: mk_test_con_with_id(1),
+                data: wire::Wire::PeerQuery(wire::PeerQuery {
+                    space: test_space(1),
+                    basis_loc: DhtLocation::new(1),
+                }),
+                respond: Box::new(|r| {
+                    async move {
+                        send_res.send(r).unwrap();
+                        ()
+                    }
+                    .boxed()
+                    .into()
+                }),
+            })
+            .await
+            .unwrap();
+
+        let response = tokio::time::timeout(Duration::from_secs(1), read_res)
+            .await
+            .expect("Timed out while waiting for a response")
+            .unwrap();
+
+        let reason = match response {
+            Wire::Failure(f) => f.reason,
+            r => panic!("Unexpected response - {:?}", r),
+        };
+
+        assert_eq!(
+            "Error querying agents: Other(\"a test error\")".to_string(),
+            reason
+        );
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
+    #[ignore = "This crashes the process because it's hitting an `unimplemented` condition"]
```

Comment:
This one is pretty bad, it will actually crash Holochain if somebody decides to send a wrong payload. The other `MetaNetEvent::` events are handling this cleanly by matching on unhandled messages and logging a warning. I think fixing this one is as simple as following the same pattern. Again, leaving the code as it is and I'll follow with a fix